### PR TITLE
ExplicitCollectionElementAccessMethod rule update

### PR DESCRIPTION
**Pull request by atulgpt**
Fixes https://github.com/detekt/detekt/issues/5640

### CODE REVIEW

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethod.kt [hunk 1]
```diff
@@ -65,6 +67,8 @@ class ExplicitCollectionElementAccessMethod(config: Config = Config.empty) : Rul
         }
         if (getter == null) return false
 
+        expression.valueArguments.drop(1).any { it.isSpread }.ifTrue { return false }
```
Comment (BraisGabin) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethod.kt:70]: Why do you drop the first one?

Comment (atulgpt) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethod.kt:70]: Hi, @BraisGabin first one would be the "key" parameter which will never be spread. Right?

Comment (BraisGabin) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethod.kt:70]: Well, that only happens in my example. But the get could only have one argument and that could be `vararg`.

Comment (atulgpt) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethod.kt]: Yup.. You are right 😅. Made those changes

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethodSpec.kt [hunk 2]
```diff
@@ -383,6 +384,105 @@ class ExplicitCollectionElementAccessMethodSpec {
...
+                }
+
+                @Test
+                fun `does report with 1 value is passed for vararg parameter as key`() {
+                    val code = """
+                        class C {
+                            operator fun get(vararg objects: Int): String = objects.toString()
+                        }
+                        fun test(c: C) {
+                            val objects = listOf(0, 1).toIntArray()
```
Comment (BraisGabin) [inline detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethodSpec.kt:479]:
```suggestion
```
This line is not needed, right?

**Review by BraisGabin — APPROVED** [review]

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethod.kt [hunk 3]
```diff
@@ -39,6 +40,7 @@ import org.jetbrains.kotlin.types.typeUtil.supertypes
  * </compliant>
  */
 @RequiresTypeResolution
+@Suppress("ReturnCount")
```
Comment (cortinico) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethod.kt:43]: Please move this suppression to the function level

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethodSpec.kt [hunk 4]
```diff
@@ -13,6 +13,7 @@ import org.junit.jupiter.api.Test
 class ExplicitCollectionElementAccessMethodSpec {
     val subject = ExplicitCollectionElementAccessMethod(Config.empty)
 
+    @Suppress("ClassName")
```
Comment (cortinico) [inline detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethodSpec.kt:16]: Why do you need this?

Comment (atulgpt) [reply detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethodSpec.kt:16]: It was giving warning for the not capital class name

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethod.kt (hunk 3 above)
Comment (atulgpt) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethod.kt:43]: I am okay with moving it to the function level but in that case, I will have to annotate two functions `isIndexGetterRecommended` and `shouldReplace`. While in the project I have seen a few cases of similar patterns of suppressing `ReturnCount` warning at a class level where there are multiple methods causing the warning. 
Let me know if I have to move it to class level still

Comment (BraisGabin) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethod.kt:43]: Yes, please, move it to the function level.

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethodSpec.kt (hunk 4 above)
Comment (atulgpt) [reply detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethodSpec.kt]: For now I have remove this `Suppress`. But we should ideally disable this rule for test files to reduce IDE warning message clutter. We can discuss this in some other issue :)

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethod.kt [hunk 5]
```diff
@@ -96,6 +100,7 @@ class ExplicitCollectionElementAccessMethod(config: Config = Config.empty) : Rul
         return function.isOperator
     }
 
+    @Suppress("ReturnCount")
```
Comment (BraisGabin) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethod.kt:103]: Why is this suppress needed? It wasn't needed before and you didn't add any new code to this function

Comment (atulgpt) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethod.kt:103]: It was already giving the warning as it has three `return` statements(while only 2 are allowed). I suppressed the warning

**Comment by atulgpt** [comment]
Hi, @BraisGabin/ @cortinico  this is good for a merge if we can close the comment thread at https://github.com/detekt/detekt/pull/5654#discussion_r1062157057. Also, want to know what is the process of the merge. Will one of the maintainers be merging this PR or if approved, the author can merge it?

**Pull request merged**